src/constants.rs

```diff
@@ -472,57 +474,102 @@ pub const RPM_MAGIC: [u8; 4] = [0xed, 0xab, 0xee, 0xdb];
+        const TRIGGERIN = 1 << 16;  // %triggerin dependency
+        const TRIGGERUN = 1 << 17;  // %triggerun dependency
+        const TRIGGERPOSTUN = 1 << 18;  // %triggerpostun dependency
+        const MISSINGOK = 1 << 19;  //suggests/enhances hint
+        const PREUNTRANS	= 1 << 20;	// %preuntrans dependency
+        const POSTUNTRANS = 1 << 21;	// %postuntrans dependency
+        // bits 22-23 unused
+        const RPMLIB = 1 << 24;	      // rpmlib(feature) dependency.
+        const TRIGGERPREIN = 1 << 25;  // %triggerprein dependency
+        const KEYRING	= 1 << 26;
+        // bit 27 unused
+        const CONFIG	= 1 << 28;
+        const META	= 1 << 29;	      // meta dependency
+    }
+}
+
+bitflags! {
+    #[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
+    pub struct FileVerifyFlags: u32 {
+        const VERIFY_NONE 	= 0;
+        const VERIFY_MD5 	= 1 << 0;	   // from %verify(md5) - obsolete */
+        const VERIFY_FILEDIGEST = 1 << 0;   // from %verify(filedigest) */
+        const VERIFY_FILESIZE 	= 1 << 1;  // from %verify(size) */
+        const VERIFY_LINKTO 	= 1 << 2;	   // from %verify(link)
+        const VERIFY_USER 	= 1 << 3;	   // from %verify(user)
+        const VERIFY_GROUP 	= 1 << 4;	   // from %verify(group)
+        const VERIFY_MTIME 	= 1 << 5;	   // from %verify(mtime)
+        const VERIFY_MODE 	= 1 << 6;	   // from %verify(mode)
+        const VERIFY_RDEV 	= 1 << 7;	   // from %verify(rdev)
+        const VERIFY_CAPS 	= 1 << 8;	   // from %verify(caps)
+        // bits 9-14 unused, reserved for rpmVerifyAttrs
+        const VERIFY_CONTEXTS	= 1 << 15;	// verify: from --nocontexts
+        // bits 16-22 used in rpmVerifyFlags
+        // bits 23-27 used in rpmQueryFlags
+        const VERIFY_READLINKFAIL= 1 << 28;	// readlink failed
+        const VERIFY_READFAIL	= 1 << 29;	// file read failed
+        const VERIFY_LSTATFAIL	= 1 << 30;	// lstat failed
+        const VERIFY_LGETFILECONFAIL	= 1 << 31;	// lgetfilecon failed
+    }
+}
+
+bitflags! {
+    #[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
+    pub struct FileFlags: u32 {
+        const CONFIG = 1;  // %%config
+        const DOC = 1 << 1;  // %%doc
+        const DONOTUSE = 1 << 2;  // %%donotuse
+        const MISSINGOK = 1 << 3;  // %%config(missingok)
+        const NOREPLACE = 1 << 4;  // %%config(noreplace)
+        // const SPECFILE = 1 << 5;  // first file in SRPM?
+        const GHOST = 1 << 6;  // %%ghost
+        const LICENSE = 1 << 7;  // %%license
+        const README = 1 << 8;  // %%readme
+        // bits 9-10 unused
+        const PUBKEY = 1 << 11;	// %%pubkey
+        const ARTIFACT	= 1 << 12;	// %%artifact
+    }
+}
+
+// ????
+// pub const RPMFILE_PARSE_EXCLUDE: u32	= (1 << 16),	// %%exclude
+// pub const RPMFILE_PARSE_DOCDIR: u32	= (1 << 17),	// from %%docdir
+// pub const RPMFILE_PARSE_DIR: u32		= (1 << 18),	// from %%dir
+// pub const RPMFILE_PARSE_SPECIALDIR: u32	= (1 << 19),	// special %%doc
```

Comment:
There's still some gaps missing but we can deal with it later.  I have no idea what these are actually for.